packages/good-design/src/core/web3/modals/BasicStyledModal.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect } from "react";
+import { Center, Text } from "native-base";
+
+import { withTheme } from "../../../theme";
+import { Image } from "../../images";
+import { Title } from "../../layout";
+import { useModal } from "../../../hooks";
+import { LinkButton } from "../../buttons/StyledLinkButton";
+import { LearnButton } from "../../buttons";
+import { learnSources } from "../../buttons/LearnButton";
+import { SpinnerCheckMark } from "../../animated";
+import BillyCelebration from "../../../assets/svg/billy-celebration.svg";
+
+const altModalTypes = ["loader", "other"];
+
+interface BasicModalProps {
+  show: boolean;
+  onClose: () => void;
+  withOverlay?: "blur" | "dark";
+  withCloseButton: boolean;
+  title: string;
+  modalStyle?: any;
+  headerStyle?: any;
+  bodyStyle?: any;
+  footerStyle?: any;
+  content?: string | JSX.Element;
+  buttonText?: string;
+  buttonAction?: () => void;
+  learnSource?: learnSources;
+  altLearnSource?: { link: string; label: string; icon: any };
+}
+
+interface CtaOrLearnModalProps extends BasicModalProps {
+  type: "cta" | "ctaX" | "learn" | "social";
+  extUrl?: string;
+  loading?: never;
+}
+
+interface AltModalProps extends BasicModalProps {
+  type: "loader";
+  loading: boolean;
+  extUrl?: never;
+}
+
+type StyledModalProps = CtaOrLearnModalProps | AltModalProps;
+
+const ModalHeader = ({ title }: { title: string }) => (
+  <Center backgroundColor="white" textAlign="center">
+    <Title fontFamily="heading" color="primary" fontSize="xl" lineHeight="110%">
+      {title}
+    </Title>
+  </Center>
+);
+
+const ModalBody = ({
+  content,
+  type,
+  loading
+}: {
+  content: string | JSX.Element | undefined;
+  type: string;
+  loading?: boolean;
+}) => (
+  <Center padding={0}>
+    {type === "loader" ? (
+      <SpinnerCheckMark loading={loading ?? true} />
+    ) : (
+      <Text color="goodGrey.600" fontFamily="subheading" lineHeight={20} padding={0} fontSize="sm">
+        {content}
+      </Text>
+    )}
+  </Center>
+);
+
+const ModalFooter = ({
+  type,
+  action,
+  extUrl,
+  buttonText,
+  source,
+  altSource = { link: "", label: "", icon: null }
+}: {
+  type: string;
+  action?: () => void;
+  extUrl?: string;
+  buttonText?: string;
+  source?: learnSources;
+  altSource?: { link: string; label: string; icon: any };
+}) => (
+  <Center padding="0" w="100%">
+    {type === "cta" && action && buttonText ? <LinkButton buttonText={buttonText} onPress={action} /> : null}
```

Comment:
it would be better to accept children so you could create dedicated modals by defining custom body / footer components and combining them

it would be easier to extend, also composition is the basic pattern in React

let's discuss it at the sync

Reply:
What I was given was a set of 5 state/variants of the same modal. This is as I understand a pattern/standard that will be used whenever a new one would be needed in a flow.

I would think that if a dedicated modal not following that pattern is needed, the BasicStyledModal is not the one to use but you would use [BasicModal](https://github.com/GoodDollar/GoodWeb3-Mono/blob/9ab1150d9b7828038a9de53766e72b2b62b7897c/packages/good-design/src/core/modals/BasicModal.tsx) directly and create an alternative styled component. The composition is done on the level of BasicModal. StyledModal is just a themed wrapper. BasicStyledModal follows the pattern of the below given default modals that will be used:

![image](https://github.com/GoodDollar/GoodWeb3-Mono/assets/6606028/2565af46-1d95-4652-90a2-f57e92d16641)

The only thing that will be replaced/added is the social bar. The image now under 'social' type would likely be part of content

Reply:
discussed, set as optional wish, to be estimated by @L03TJ3 
could be resolved

Reply:
I have split up the footer component. Rest of the composition and any alternative modal can be created separately from this by using BasicModal directly (not the styled one)

No more estimation to be done from my pov